pkg/metrics/metrics.go

```diff
@@ -256,25 +215,60 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	return true
 }
 
-// updateServiceMonitors patches the metrics Service Monitor to include required fields to display node graphs on the
-// OpenShift console. Console graph queries require metrics endpoint target name to be node name, however
-// windows_exporter returns node IP. We replace the target name by adding `replace` action field to the ServiceMonitor
-// object that replaces node IP to node name as the metrics endpoint target.
-func updateServiceMonitors(cfg *rest.Config, namespace string) error {
-
-	patchData := fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/spec/endpoints/0\", "+
-		"\"value\":{\"path\": \"/%s\",\"port\": \"%s\",\"relabelings\": [{\"action\": \"replace\", \"regex\": \"(.*)\", "+
-		"\"replacement\": \"$1\", \"sourceLabels\": [\"__meta_kubernetes_endpoint_address_target_name\"],"+
-		"\"targetLabel\": \"instance\"}]}}]", PortName, PortName)
+// Configure takes care of all the required configuration steps
+// for Prometheus monitoring like validating monitoring label
+// and creating metrics Endpoints object.
+func (c *Config) Configure(ctx context.Context) error {
+	// validate if cluster monitoring is enabled in the operator namespace
+	if err := c.validate(ctx); err != nil {
+		log.Error(err, "error validating cluster monitoring label")
+		return nil
+	}
+	// Create Metrics Endpoint object in the operator namespace
+	if err := c.createEndpoint(); err != nil {
+		return errors.Wrap(err, "error creating metrics Endpoint")
+	}
```

Comment:
I thought we decided not to create the endpoint if metrics was not enabled. Or am I mistaken?

Reply:
@aravindhp we are not creating endpoints here if the validation fails, according to the implementation here if there is an error we are logging and returning nil, it could be done in another way where we check for the metricsEnabled bool value and proceed.

Reply:
My point is that you cannot assume that all errors returned by `validate()` is because metrics is disabled. What if there is an error getting the operator namespace? That cannot be treated as metrics being disabled. So either a bool needs to be returned or a specific error that indicates that metrics is disabled.